File: backend/app/controllers/notification_controller.py
class NotificationController:
    @staticmethod
    async def get_notifications(
        user_id: int, skip: int = 0, limit: int = 10
    ) -> List[NotificationSchema]:
        """
        Retrieve user notifications
        """
        try:
            notifications = await NotificationService.get_user_notifications(
                user_id, skip, limit
            )
            return notifications
        except Exception as e:
            raise ValueError(f"Error retrieving notifications: {str(e)}")

    @staticmethod
    async def mark_notification_as_read(user_id: int, notification_id: int) -> Dict:
        """
        Mark a specific notification as read
        """
        try:
            result = await NotificationService.mark_notification_read(
                user_id, notification_id
            )
            return result
        except Exception as e:
            raise ValueError(f"Error marking notification: {str(e)}")

File: backend/src/controllers/notificationController.js
const { NotificationService } = require("../services");
const { successResponse, errorResponse } = require("../utils/responseHandler");
const asyncHandler = require("../utils/asyncHandler");

class NotificationController {
  getNotifications = asyncHandler(async (req, res) => {
    const notifications = await NotificationService.getUserNotifications(
      req.user.id
    );
    successResponse(res, notifications);
  });

  markAsRead = asyncHandler(async (req, res) => {
    await NotificationService.markNotificationsAsRead(
      req.user.id,
      req.body.ids
    );
    successResponse(res, { message: "Notifications marked as read" });
  });

  sendNotification = asyncHandler(async (req, res) => {
    const { title, message, recipients } = req.body;
    await NotificationService.sendNotification(title, message, recipients);
    successResponse(res, { message: "Notification sent successfully" });
  });
}

module.exports = new NotificationController();
